### PR TITLE
Add support for allowDuplicate suggestion

### PR DESCRIPTION
**Pull request by yidea**
- add support for allowDuplicate suggestion options (tweaking for new site design implementation)
- only pass needed data in custom event typeahead:cursorMoved
- update prototype playground2.html functionality to be similiar to design requirement

@ryan-roemer @alexlande

### CODE REVIEW

File: test/dataset_spec.js [hunk 1]
```diff
@@ -395,6 +398,29 @@ describe('Dataset', function() {
       });
     });
 
+    it('does not dedup the items when allowDuplicate is set true', function () {
+      var spy = jasmine.createSpy(),
+          remote = [fixtureDatums[0], fixtureStrings[2]];
```
Comment (ryan-roemer) [inline test/dataset_spec.js:403]: What's alignment for trailing vars? If two, then please fix globally.

**Comment by ryan-roemer** [comment]
One minor comment. Rest LGTM. Approved (but you do need to ping me to merge ;) /cc @yidea

File: test/dataset_spec.js (hunk 1 above)
Comment (yidea) [inline test/dataset_spec.js:403]: it's four

**Comment by ryan-roemer** [comment]
Approved. /cc @yidea

**Pull request merged**